TeamsBridgeApp.ts

```diff
@@ -93,22 +96,39 @@ export class TeamsBridgeApp
       IPreMessageDeletePrevent,
       IPreFileUpload,
       IPreRoomUserLeave {
-    private selectedTeamsUserIds: Array<string>;
-    private changeTeamsUserMemberRoom: IRoom | undefined;
 
     constructor(info: IAppInfo, logger: ILogger, accessors: IAppAccessors) {
       super(info, logger, accessors);
     }
 
-    public async onSettingUpdated(
-      setting: ISetting,
-      configurationModify: IConfigurationModify,
-      read: IRead,
-      http: IHttp,
+    async getSettingValueById(id: string) {
+        return this.getAccessors().environmentReader.getSettings().getValueById(id)
+    }
+
+    async onInstall(
+        context: IAppInstallationContext,
+        read: IRead,
+        http: IHttp,
+        persistence: IPersistence,
+        modify: IModify
     ): Promise<void> {
-      console.log(
-        `onSettingUpdated for setting ${setting.id} with new value ${setting.value}`,
-      );
+        await createWebhookSecret({ persistence });
+    }
+
+    async onEnable(environment: IEnvironmentRead, configurationModify: IConfigurationModify): Promise<boolean> {
+        try {
+            await configurationModify.scheduler.scheduleOnce({
+                id: WebhookSecretCreationJobId,
+                when: new Date(),
+            });
+        } catch (e) {
+            this.getLogger().error(e);
+        }
+        return true;
```

Comment:
Do we really need to return a boolean here? If so, if an error occurs, should we still return true?

Reply:
The return value tells if the app should be enabled or not. The onEnable function just schedules a job to check for webhook secret presence. I dont think the schedule function would throw error. But, to be double sure that the app is enable in case of error in scheduling the job. The webhook secret is eventually created in the recurring registration renewal service.